internal/hcl/module.go

```diff
@@ -58,34 +56,8 @@ func (b *Block) getModuleBlocks(modulePath string) (Blocks, error) {
 	return blocks, err
 }
 
-// ModulesMetadata is a struct that represents the JSON found in the modules.json file int the .terraform dir.
-type ModulesMetadata struct {
-	Modules []ModuleMetadata `json:"Modules"`
-}
-
-type ModuleMetadata struct {
-	Key     string `json:"Key"`
-	Source  string `json:"Source"`
-	Version string `json:"Version"`
-	Dir     string `json:"Dir"`
-}
-
-func loadModuleMetadata(fullPath string) (*ModulesMetadata, error) {
-	metadataPath := filepath.Join(fullPath, ".terraform/modules/modules.json")
-	if _, err := os.Stat(metadataPath); err != nil {
-		return nil, fmt.Errorf("metadata file does not exist %w", err)
-	}
-
-	f, err := os.Open(metadataPath)
-	if err != nil {
-		return nil, fmt.Errorf("could not read metadata file %w", err)
-	}
-	defer func() { _ = f.Close() }()
-
-	var metadata ModulesMetadata
-	if err := json.NewDecoder(f).Decode(&metadata); err != nil {
-		return nil, fmt.Errorf("could not decode metadata file %w", err)
-	}
+func loadModules(fullPath string) (*modules.Manifest, error) {
+	moduleLoader := modules.NewModuleLoader(fullPath)
 
-	return &metadata, nil
+	return moduleLoader.Load()
 }
```

Comment:
I'd probably remove this `loadModules` function and just set the `modules.Loader` on the `Parser` as a struct. Then move the `fullPath` to the `Load` method. And whilst we're doing this let's move the `initialPath` variable to a argument on `ParseDirectory`, doesn't make a lot of sense to have it at the top level.

```
func New(initialPath string, options ...Option) *Parser {
	p := &Parser{
		workspaceName: "default",
		moduleLoader: modules.NewModuleLoader(),
	}
        ...
}


func (parser *Parser) ParseDirectory(initialPath string) ([]*Module, error) {
      ....
      modulesMetadata, err := p.moduleLoader.Load(initialPath)
      ...
}
```

Easier for unit testing in the future

Reply:
The `ModuleLoader` creates a cache which should be unique per initial path. Would it make sense to construct that each time `Load` is called as well if we're moving the path to be an argument of that function? My only concern with this is that it makes the code a bit messier having to pass the path and cache to each function in the `ModuleLoader`. But if I get rid of the `downloadDir` and `manifestFilePath` functions then it's probably alright.

Reply:
spoken about this over zoom and decision is that we leave as is as this the way Terraform does it with. Further docs to be added.